src/Sarif/ArtifactProvider.cs

```diff
@@ -25,108 +24,4 @@ public ArtifactProvider(IEnumerable<IEnumeratedArtifact> artifacts)
 
         public IFileSystem FileSystem { get; set; }
     }
-
-    public class SinglethreadedZipArchiveArtifactProvider : ArtifactProvider
```

Comment:
>SinglethreadedZipArchiveArtifactProvider [](http://example.com/codeflow?start=17&length=40)

Notoriously, this single file has included multiple class definitions, I finally took a few minutes and extracted all of them.